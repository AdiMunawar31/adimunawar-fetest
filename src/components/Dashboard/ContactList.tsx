import React, { useState } from "react";
import { AiFillCloseCircle } from "react-icons/ai";
import { BsThreeDotsVertical } from "react-icons/bs";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import { asyncDeleteContact } from "../../states/deleteContact/action";
import Swal from "sweetalert2";
import { Contact } from "../../types/ContactTypes";

const ContactList: React.FC<Contact> = ({
  id,
  firstName,
  lastName,
  age,
  photo,
}) => {
  const [isPopupVisible, setIsPopupVisible] = useState(false);
  const dispatch: any = useDispatch();
  const navigate: any = useNavigate();

  const togglePopup = () => {
    setIsPopupVisible(!isPopupVisible);
  };

  const handleChat = () => {
    navigate(`/chat/${id}`);
  };

  const handleDelete = async () => {
    const confirmationResult = await Swal.fire({
      title: "Are you sure?",
      text: "You won't be able to delete this contact!",
      icon: "error",
      showCancelButton: true,
      confirmButtonColor: "#3085d6",
      cancelButtonColor: "#d33",
      confirmButtonText: "Yes, delete it!",
      background: "#edfdff",
    });

    if (confirmationResult.isConfirmed) {
      dispatch(asyncDeleteContact(id));
      setIsPopupVisible(false);
      navigate("/dashboard");

      Swal.fire({
        title: "Deleted!",
        text: "Your Contact has been deleted.",
        icon: "success",
        background: "#edfdff",
      }).then(() => {});
    }
  };

  const handleEdit = () => {
    navigate(`/edit/${id}`);
    setIsPopupVisible(false);
  };

  return (
    <li className='border-b cursor-pointer contact-item' key={`${id}-contact`}>
      <div className='flex items-center justify-between w-full px-4 py-0.5 select-none hover:bg-gray-100 focus:outline-none relative'>
        <div className='flex pr-20' onClick={handleChat}>
          <img
            className='w-12 mr-3 h-12 border-cyan-500 rounded-full border-2 p-0.5'
            src={
              photo !== "N/A"
                ? photo
                : `https://ui-avatars.com/api/?name=${firstName} ${lastName}/`
            }
            alt='Photo profile'
          />
          <div className='block'>
            <h3 className='font-semibold capitalize -mb-1 truncate w-40'>
              {firstName} {lastName}
            </h3>
            <span className='text-xs text-gray-500'>Age: {age}</span>
          </div>
        </div>
        <button
          onClick={togglePopup}
          className='p-2 top-1/2 transform mt-8 -translate-y-1/2'
        >
          {isPopupVisible ? (
            <AiFillCloseCircle className='text-red-400' size={20} />
          ) : (
            <BsThreeDotsVertical />
          )}
        </button>

        {isPopupVisible && (
          <div className='absolute right-12 top-1/2 px-2 py-1 transform -translate-y-1/2 w-32 bg-white border rounded-lg shadow-lg'>
            <button
              onClick={handleEdit}
              className='block w-full text-left py-1 px-6 hover:bg-gray-100 focus:outline-none'
            >
              Edit
            </button>
            <button
              className='block w-full text-left py-1 px-6 hover-bg-gray-100 focus:outline-none'
              onClick={handleDelete}
            >
              Delete
            </button>
          </div>
        )}
      </div>
    </li>
  );
};

export default ContactList;
