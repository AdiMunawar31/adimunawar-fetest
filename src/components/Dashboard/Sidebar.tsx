import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { RiShutDownLine } from "react-icons/ri";
import { asyncPopulateContacts } from "../../states/getContact/action";
import ContactList from "./ContactList";
import { SidebarProps } from "../../types/SidebarProps";
import logoImage from "/contact.png";
import { Contact } from "../../types/ContactTypes";
import { ContactState } from "../../types/ContactState";

const Sidebar: React.FC<SidebarProps> = ({ children }) => {
  const dispatch: any = useDispatch();
  const contacts = useSelector((state: ContactState) => state.contacts || []);

  useEffect(() => {
    dispatch(asyncPopulateContacts());
  }, [dispatch]);

  return (
    <div className='bg-gradient-to-br from-cyan-300 via-cyan-100 to-blue-900 w-screen h-screen sm:p-5'>
      <div className='bg-white border border-gray-200 rounded flex h-full'>
        <div className='w-full sm:w-1/2 md:w-1/3 lg:w-1/4 h-full overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch'>
          <div className='border-b border-gray-200 p-3 relative flex justify-between'>
            <button className='flex items-start select-none font-semibold focus:outline-none'>
              <Link to='/dashboard' className='flex'>
                <img src={logoImage} alt='logo' width={40} />
                <span className='text-lg text-cyan-400 font-bold ml-2 mt-2'>
                  DiContact
                </span>
              </Link>
            </button>
            <Link
              to='/'
              className='bg-red-200 p-2 flex items-center rounded-md'
              title='Shutdown'
            >
              <RiShutDownLine className='text-red-500 text-xl font-bold' />
            </Link>
          </div>

          <div className='flex items-center justify-between text-sm border-b border-gray-200'>
            <div>
              <button className='py-3 w-20 uppercase font-semibold select-none h-full focus:outline-none border-b border-black'>
                Primary
              </button>
              <button className='py-3 w-20 uppercase text-gray-400 select-none h-full focus:outline-none border-b border-transparent'>
                General
              </button>
            </div>
            <button className='p-3 font-semibold text-blue-500 select-none h-full focus:outline-none border-b border-transparent'>
              {contacts.length} Contacts
            </button>
          </div>

          <ul className='py-1'>
            {contacts
              .slice()
              .reverse()
              .map((contact: Contact) => (
                <ContactList key={contact.id} {...contact} />
              ))}
          </ul>
        </div>

        <div className='hidden sm:w-1/2 md:w-2/3 lg:w-3/4 border-l border-gray-200 sm:flex items-center justify-center text-center overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch'>
          {children}
        </div>
      </div>
    </div>
  );
};

export default Sidebar;
