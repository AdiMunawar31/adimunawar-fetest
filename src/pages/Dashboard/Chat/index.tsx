import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import Sidebar from "../../../components/Dashboard/Sidebar";
import { asyncReceiveContactDetail } from "../../../states/getDetailContact/action";
import { Contact } from "../../../types/ContactTypes";

const Chat = () => {
  const { contactId } = useParams();

  const contactDetail: Contact = useSelector(
    (state: any) => state.contactDetail
  );

  const { firstName, lastName, photo, age } = contactDetail;

  const dispatch: any = useDispatch();

  useEffect(() => {
    dispatch(asyncReceiveContactDetail(contactId));
  }, [contactId, dispatch]);

  return (
    <Sidebar>
      <div className='flex-1 p:2 sm:p-6 justify-between flex flex-col h-screen'>
        <div className='flex sm:items-center justify-between py-3 border-b-2 border-gray-200'>
          <div className='relative flex items-center space-x-4'>
            <div className='relative'>
              <span className='absolute text-green-400 right-0 bottom-0'>
                <svg width={20} height={20}>
                  <circle cx={8} cy={8} r={8} fill='currentColor' />
                </svg>
              </span>
              <img
                src={
                  photo != "N/A"
                    ? photo
                    : `https://ui-avatars.com/api/?name=${firstName} ${lastName}/`
                }
                alt='photo profile'
                className='w-10 sm:w-16 h-10 sm:h-16 rounded-full'
              />
            </div>
            <div className='-mt-2'>
              <span className='text-lg mt-1 text-gray-700 font-semibold'>
                {firstName} {lastName}
              </span>
              <span className='text-sm text-gray-600 ml-2'>
                {"( Software Dev )"}
              </span>
            </div>
          </div>
          <div className='flex items-center space-x-2'>
            <button
              type='button'
              className='inline-flex items-center justify-center rounded-lg border h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
            >
              <svg
                xmlns='http://www.w3.org/2000/svg'
                fill='none'
                viewBox='0 0 24 24'
                stroke='currentColor'
                className='h-6 w-6'
              >
                <path
                  strokeLinecap='round'
                  strokeLinejoin='round'
                  strokeWidth={2}
                  d='M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z'
                />
              </svg>
            </button>
            <button
              type='button'
              className='inline-flex items-center justify-center rounded-lg border h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
            >
              <svg
                xmlns='http://www.w3.org/2000/svg'
                fill='none'
                viewBox='0 0 24 24'
                stroke='currentColor'
                className='h-6 w-6'
              >
                <path
                  strokeLinecap='round'
                  strokeLinejoin='round'
                  strokeWidth={2}
                  d='M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z'
                />
              </svg>
            </button>
            <button
              type='button'
              className='inline-flex items-center justify-center rounded-lg border h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
            >
              <svg
                xmlns='http://www.w3.org/2000/svg'
                fill='none'
                viewBox='0 0 24 24'
                stroke='currentColor'
                className='h-6 w-6'
              >
                <path
                  strokeLinecap='round'
                  strokeLinejoin='round'
                  strokeWidth={2}
                  d='M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9'
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          id='messages'
          className='flex flex-col h-full justify-start  mt-4 text-start space-y-4 p-3 overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch'
        >
          <div className='chat-message'>
            <div className='flex items-end'>
              <div className='flex flex-col space-y-2 text-sm max-w-xs mx-2 order-2 items-start'>
                <div>
                  <span className='px-4 py-2 rounded-lg inline-block rounded-bl-none bg-gray-300 text-gray-600'>
                    <p>
                      Hello, I'm{" "}
                      <b>
                        {firstName} {lastName}
                      </b>
                    </p>
                    <p>
                      I'm <b>{age}</b> years old
                    </p>
                    <p>
                      I work as a <b>Software Dev</b>
                    </p>
                  </span>
                </div>
              </div>
              <img
                src={
                  photo != "N/A"
                    ? photo
                    : `https://ui-avatars.com/api/?name=${firstName} ${lastName}/`
                }
                alt='photo profile'
                className='w-6 h-6 rounded-full order-1'
              />
            </div>
          </div>
          <div className='chat-message'>
            <div className='flex items-end justify-end'>
              <div className='flex flex-col space-y-2 text-sm max-w-xs mx-2 order-1 items-end'>
                <div>
                  <span className='px-4 py-2 rounded-lg inline-block rounded-br-none bg-cyan-500 text-white '>
                    Welcome to {firstName} {lastName}! ðŸŽ‰ We're delighted that
                    you've joined our community. Feel free to explore our
                    fantastic features and start connecting with new friends. If
                    you have any questions, don't hesitate to ask. Enjoy your
                    time!
                  </span>
                </div>
              </div>
              <img
                src='/contact.png'
                alt='My profile'
                className='w-6 h-6 rounded-full order-2'
              />
            </div>
          </div>
        </div>
        <div className='border-t-2 pb-3 border-gray-200 px-4 pt-4 mb-2 sm:mb-0'>
          <div className='relative flex'>
            <span className='absolute inset-y-0 flex items-center'>
              <button
                type='button'
                className='inline-flex items-center justify-center rounded-full h-12 w-12 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
              >
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill='none'
                  viewBox='0 0 24 24'
                  stroke='currentColor'
                  className='h-6 w-6 text-gray-600'
                >
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    strokeWidth={2}
                    d='M19 11a7 7 0 01-7 7m0 0a7 7 0 01-7-7m7 7v4m0 0H8m4 0h4m-4-8a3 3 0 01-3-3V5a3 3 0 116 0v6a3 3 0 01-3 3z'
                  />
                </svg>
              </button>
            </span>
            <input
              type='text'
              placeholder='Write your message!'
              className='w-full focus:outline-none focus:placeholder-gray-400 text-gray-600 placeholder-gray-600 pl-12 bg-gray-200 rounded-md py-3'
            />
            <div className='absolute right-0 items-center inset-y-0 hidden sm:flex'>
              <button
                type='button'
                className='inline-flex items-center justify-center rounded-full h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
              >
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill='none'
                  viewBox='0 0 24 24'
                  stroke='currentColor'
                  className='h-6 w-6 text-gray-600'
                >
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    strokeWidth={2}
                    d='M15.172 7l-6.586 6.586a2 2 0 102.828 2.828l6.414-6.586a4 4 0 00-5.656-5.656l-6.415 6.585a6 6 0 108.486 8.486L20.5 13'
                  />
                </svg>
              </button>
              <button
                type='button'
                className='inline-flex items-center justify-center rounded-full h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
              >
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill='none'
                  viewBox='0 0 24 24'
                  stroke='currentColor'
                  className='h-6 w-6 text-gray-600'
                >
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    strokeWidth={2}
                    d='M3 9a2 2 0 012-2h.93a2 2 0 001.664-.89l.812-1.22A2 2 0 0110.07 4h3.86a2 2 0 011.664.89l.812 1.22A2 2 0 0018.07 7H19a2 2 0 012 2v9a2 2 0 01-2 2H5a2 2 0 01-2-2V9z'
                  />
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    strokeWidth={2}
                    d='M15 13a3 3 0 11-6 0 3 3 0 016 0z'
                  />
                </svg>
              </button>
              <button
                type='button'
                className='inline-flex items-center justify-center rounded-full h-10 w-10 transition duration-500 ease-in-out text-gray-500 hover:bg-gray-300 focus:outline-none'
              >
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill='none'
                  viewBox='0 0 24 24'
                  stroke='currentColor'
                  className='h-6 w-6 text-gray-600'
                >
                  <path
                    strokeLinecap='round'
                    strokeLinejoin='round'
                    strokeWidth={2}
                    d='M14.828 14.828a4 4 0 01-5.656 0M9 10h.01M15 10h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z'
                  />
                </svg>
              </button>
              <button
                aria-label='Send'
                type='button'
                className='inline-flex items-center justify-center rounded-lg px-4 py-3 transition duration-500 ease-in-out text-white bg-blue-900 hover:bg-blue-700 focus:outline-none'
              >
                <span className='font-bold'>Send</span>
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  viewBox='0 0 20 20'
                  fill='currentColor'
                  className='h-6 w-6 ml-2 transform rotate-90'
                >
                  <path d='M10.894 2.553a1 1 0 00-1.788 0l-7 14a1 1 0 001.169 1.409l5-1.429A1 1 0 009 15.571V11a1 1 0 112 0v4.571a1 1 0 00.725.962l5 1.428a1 1 0 001.17-1.408l-7-14z' />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </Sidebar>
  );
};

export default Chat;
