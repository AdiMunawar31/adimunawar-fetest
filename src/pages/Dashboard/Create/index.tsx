import { yupResolver } from "@hookform/resolvers/yup";
import { FormValues } from "@src/types/FormValuesTypes";
import React, { useEffect, useState } from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import { MdContacts } from "react-icons/md";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import * as Yup from "yup";
import Sidebar from "../../../components/Dashboard/Sidebar";
import InputField from "../../../components/InputField";
import { asyncAddContact } from "../../../states/addContact/action";

const schema = Yup.object().shape({
  firstName: Yup.string().required("First Name is required"),
  lastName: Yup.string().required("Last Name is required"),
  age: Yup.number()
    .required("Age is required")
    .positive("Age must be positive"),
  photo: Yup.string()
    .url("Enter a valid URL")
    .required("Photo URL is required"),
});

const CreateContact: React.FC = () => {
  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
  const dispatch: any = useDispatch();

  const {
    register,
    handleSubmit,
    control,
    formState: { errors, isDirty },
    reset,
  } = useForm<FormValues>({
    resolver: yupResolver(schema),
  });

  const onSubmit: SubmitHandler<FormValues> = (data) => {
    const trimmedData = {
      ...data,
      age: data.age,
      firstName: data.firstName.trim(),
      lastName: data.lastName.trim(),
      photo: data.photo.trim(),
    };

    dispatch(asyncAddContact(trimmedData));
    reset();
  };

  useEffect(() => {
    setIsSubmitDisabled(!isDirty);
  }, [isDirty]);

  return (
    <Sidebar>
      <div className='container mx-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch'>
        <div className='w-full bg-white p-5 rounded-md'>
          <div className='flex mt-8 text-lg mx-6 text-start p-5 bg-blue-50 font-semibold text-cyan-500'>
            <MdContacts className='mt-1 mr-3' />
            <span>Add Contact Data Form</span>
          </div>
          <div className='m-7'>
            <form onSubmit={handleSubmit(onSubmit)}>
              <InputField
                control={control}
                label='First Name'
                name='firstName'
                type='text'
                placeholder='e.g John'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Last Name'
                name='lastName'
                type='text'
                placeholder='e.g Doe'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Age'
                name='age'
                type='number'
                placeholder='e.g 31'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Photo'
                name='photo'
                type='text'
                placeholder='e.g https://example-photos.png'
                register={register}
                errors={errors}
              />
              <div className='w-4/12 gap-2 my-8 flex'>
                <button
                  type='submit'
                  disabled={isSubmitDisabled}
                  className={`w-full py-2 px-3 text-white rounded-md focus:bg-blue-600 focus:outline-none ${
                    isSubmitDisabled
                      ? "bg-gray-400 cursor-not-allowed"
                      : "bg-blue-900 hover:bg-blue-600"
                  }`}
                >
                  Add Contact
                </button>
                <Link
                  to='/dashboard'
                  className={`w-full py-2 px-3 text-white rounded-md focus:bg-red-600 focus:outline-none bg-red-500 hover:bg-red-400`}
                >
                  Cancel
                </Link>
              </div>
            </form>
          </div>
        </div>
      </div>
    </Sidebar>
  );
};

export default CreateContact;
