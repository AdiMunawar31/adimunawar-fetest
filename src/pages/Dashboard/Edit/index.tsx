import { yupResolver } from "@hookform/resolvers/yup";
import React, { useEffect, useState } from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import { MdContacts } from "react-icons/md";
import { useDispatch, useSelector } from "react-redux";
import { Link, useNavigate, useParams } from "react-router-dom";
import * as Yup from "yup";
import Sidebar from "../../../components/Dashboard/Sidebar";
import InputField from "../../../components/InputField";
import { asyncEditContact } from "../../../states/editContact/action";
import { asyncReceiveContactDetail } from "../../../states/getDetailContact/action";
import Swal from "sweetalert2";

interface FormValues {
  firstName: string;
  lastName: string;
  age: number;
  photo: string;
}

const schema = Yup.object().shape({
  firstName: Yup.string().required("First Name is required"),
  lastName: Yup.string().required("Last Name is required"),
  age: Yup.number()
    .required("Age is required")
    .positive("Age must be positive"),
  photo: Yup.string()
    .url("Enter a valid URL")
    .required("Photo URL is required"),
});

const EditContact: React.FC = () => {
  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
  const dispatch: any = useDispatch();
  const { contactId } = useParams();
  const navigate = useNavigate();

  const state: any = useSelector((states) => states);
  const { id, firstName, lastName, photo, age } = state.contactDetail;

  const {
    register,
    handleSubmit,
    control,
    formState: { errors, isDirty },
    setValue,
  } = useForm<FormValues>({
    resolver: yupResolver(schema),
  });

  const onSubmit: SubmitHandler<FormValues> = (data) => {
    const trimmedData = {
      ...data,
      firstName: data.firstName.trim(),
      lastName: data.lastName.trim(),
      photo: data.photo.trim(),
    };

    dispatch(asyncEditContact(trimmedData, id));
    Swal.fire({
      title: "Updated!",
      text: "Your Contact has been updated.",
      icon: "success",
      background: "#edfdff",
    }).then(() => {
      navigate("/dashboard");
    });
  };

  useEffect(() => {
    dispatch(asyncReceiveContactDetail(contactId));
    setValue("firstName", firstName);
    setValue("lastName", lastName);
    setValue("age", age);
    setValue("photo", photo);
  }, [age, contactId, dispatch, firstName, lastName, photo, setValue]);

  useEffect(() => {
    setIsSubmitDisabled(!isDirty);
  }, [isDirty]);

  return (
    <Sidebar>
      <div className='container mx-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch'>
        <div className='w-full bg-white p-5 rounded-md'>
          <div className='flex mt-8 text-lg mx-6 text-start p-5 bg-blue-50 font-semibold text-cyan-500'>
            <MdContacts className='mt-1 mr-3' />
            <span>Edit Contact Data Form</span>
          </div>
          <div className='m-7'>
            <form onSubmit={handleSubmit(onSubmit)}>
              <InputField
                control={control}
                label='First Name'
                name='firstName'
                type='text'
                placeholder='e.g John'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Last Name'
                name='lastName'
                type='text'
                placeholder='e.g Doe'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Age'
                name='age'
                type='number'
                placeholder='e.g 31'
                register={register}
                errors={errors}
              />
              <InputField
                control={control}
                label='Photo'
                name='photo'
                type='text'
                placeholder='e.g https://example-photos.png'
                register={register}
                errors={errors}
              />
              <div className='w-4/12 gap-2 my-8 flex'>
                <button
                  type='submit'
                  disabled={isSubmitDisabled}
                  className={`w-full py-2 px-3 text-white rounded-md focus:bg-blue-600 focus:outline-none ${
                    isSubmitDisabled
                      ? "bg-gray-400 cursor-not-allowed"
                      : "bg-blue-900 hover:bg-blue-600"
                  }`}
                >
                  Edit Contact
                </button>
                <Link
                  to='/dashboard'
                  className={`w-full py-2 px-3 text-white rounded-md focus:bg-red-600 focus:outline-none bg-red-500 hover:bg-red-400`}
                >
                  Cancel
                </Link>
              </div>
            </form>
          </div>
        </div>
      </div>
    </Sidebar>
  );
};

export default EditContact;
