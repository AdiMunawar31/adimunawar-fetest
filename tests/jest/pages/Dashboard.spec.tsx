import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Dashboard from "@pages/Dashboard";

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn(),
  useSelector: () => [
    {
      id: "1",
      firstName: "John",
      lastName: "Doe",
      photo: "john.png",
    },
    {
      id: "2",
      name: "Jane",
      lastName: "Smith",
      photo: "janesmith.png",
    },
  ],
}));

describe("Dashboard component", () => {
  it("renders the Dashboard component", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={["/dashboard"]}>
        <Dashboard />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
