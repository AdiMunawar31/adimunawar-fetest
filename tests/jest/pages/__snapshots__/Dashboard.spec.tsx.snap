// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the Dashboard component 1`] = `
<div
  className="bg-gradient-to-br from-cyan-300 via-cyan-100 to-blue-900 w-screen h-screen sm:p-5"
>
  <div
    className="bg-white border border-gray-200 rounded flex h-full"
  >
    <div
      className="w-full sm:w-1/2 md:w-1/3 lg:w-1/4 h-full overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch"
    >
      <div
        className="border-b border-gray-200 p-3 relative flex justify-between"
      >
        <button
          className="flex items-start select-none font-semibold focus:outline-none"
        >
          <a
            className="flex"
            href="/dashboard"
            onClick={[Function]}
          >
            <img
              alt="logo"
              src="/contact.png"
              width={40}
            />
            <span
              className="text-lg text-cyan-400 font-bold ml-2 mt-2"
            >
              DiContact
            </span>
          </a>
        </button>
        <a
          className="bg-red-200 p-2 flex items-center rounded-md"
          href="/"
          onClick={[Function]}
          title="Shutdown"
        >
          <svg
            className="text-red-500 text-xl font-bold"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.26489 3.80698L7.41191 5.44558C5.34875 6.89247 4 9.28873 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 9.28873 18.6512 6.89247 16.5881 5.44558L17.7351 3.80698C20.3141 5.61559 22 8.61091 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 8.61091 3.68594 5.61559 6.26489 3.80698ZM11 12V2H13V12H11Z"
            />
          </svg>
        </a>
      </div>
      <div
        className="flex items-center justify-between text-sm border-b border-gray-200"
      >
        <div>
          <button
            className="py-3 w-20 uppercase font-semibold select-none h-full focus:outline-none border-b border-black"
          >
            Primary
          </button>
          <button
            className="py-3 w-20 uppercase text-gray-400 select-none h-full focus:outline-none border-b border-transparent"
          >
            General
          </button>
        </div>
        <button
          className="p-3 font-semibold text-blue-500 select-none h-full focus:outline-none border-b border-transparent"
        >
          2
           Contacts
        </button>
      </div>
      <ul
        className="py-1 contact-list"
      >
        <li
          className="border-b cursor-pointer contact-item"
        >
          <div
            className="flex items-center justify-between w-full px-4 py-0.5 select-none hover:bg-gray-100 focus:outline-none relative"
          >
            <div
              className="flex pr-20"
              onClick={[Function]}
            >
              <img
                alt="Photo profile"
                className="w-12 mr-3 h-12 border-cyan-500 rounded-full border-2 p-0.5"
                src="janesmith.png"
              />
              <div
                className="block"
              >
                <h3
                  className="font-semibold capitalize -mb-1 truncate w-40"
                >
                   
                  Smith
                </h3>
                <span
                  className="text-xs text-gray-500"
                >
                  Age: 
                </span>
              </div>
            </div>
            <button
              className="p-2 top-1/2 transform mt-8 -translate-y-1/2"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          className="border-b cursor-pointer contact-item"
        >
          <div
            className="flex items-center justify-between w-full px-4 py-0.5 select-none hover:bg-gray-100 focus:outline-none relative"
          >
            <div
              className="flex pr-20"
              onClick={[Function]}
            >
              <img
                alt="Photo profile"
                className="w-12 mr-3 h-12 border-cyan-500 rounded-full border-2 p-0.5"
                src="john.png"
              />
              <div
                className="block"
              >
                <h3
                  className="font-semibold capitalize -mb-1 truncate w-40"
                >
                  John
                   
                  Doe
                </h3>
                <span
                  className="text-xs text-gray-500"
                >
                  Age: 
                </span>
              </div>
            </div>
            <button
              className="p-2 top-1/2 transform mt-8 -translate-y-1/2"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
                />
              </svg>
            </button>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="hidden sm:w-1/2 md:w-2/3 lg:w-3/4 border-l border-gray-200 sm:flex items-center justify-center text-center overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch"
    >
      <div
        className="space-y-5"
      >
        <div
          className="border border-black rounded-full inline-flex p-5 items-center justify-center"
        >
          <svg
            className="transform translate-y-1"
            height={52}
            viewBox="0 0 48 48"
            width={52}
          >
            <path
              d="M47.8 3.8c-.3-.5-.8-.8-1.3-.8h-45C.9 3.1.3 3.5.1 4S0 5.2.4 5.7l13.2 13c.5.4 1.1.6 1.7.3l16.6-8c.7-.3 1.6-.1 2 .5.4.7.2 1.6-.5 2l-15.6 9.9c-.5.3-.8 1-.7 1.6l4.6 19c.1.6.6 1 1.2 1.1h.2c.5 0 1-.3 1.3-.7l23.2-39c.5-.5.5-1.1.2-1.6z"
            />
          </svg>
        </div>
        <div
          className="space-y-0.5"
        >
          <h1
            className="font-semibold text-xl"
          >
            Your Messages
          </h1>
          <p
            className="text-gray-600 min-w-46"
          >
            Send private photos and messages to a friend or group
          </p>
        </div>
        <button
          className="bg-blue-500 py-1 px-3 rounded text-white select-none focus:outline-none"
        >
          <a
            href="/create-contact"
            onClick={[Function]}
          >
            Add Contact
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
